### ISSUE TITLE
Angabe einer Hood optional machen

### ISSUE THREAD

**Issue by fblaese** [asdil12/ff-map#221]
Layer 3 Router, die keine Hood mitsenden, sollten nicht in der Hood "Legacy" landen. Entsprechend sollten sie auch nicht im "Routers V1" Layer auftauchen.

**Comment by adschm** [comment]
Wie willst du unterscheiden, ob es sich um einen V1 Router oder einen Layer-3 Router handelt? Soweit ich mich erinnere gibt es keine robuste Möglichkeit, dies zu unterscheiden ...

Man könnte höchstens einfach den Namen ändern, da V1 ja inzwischen irrelevant ist. Dann kommen halt alle L3 ohne Hood und V1 in die Hood "No Hood"?

**Comment by fblaese** [comment]
> Wie willst du unterscheiden, ob es sich um einen V1 Router oder einen Layer-3 Router handelt? Soweit ich mich erinnere gibt es keine robuste Möglichkeit, dies zu unterscheiden ...

Nicht, wir nehmen keine Daten von v1 Routern mehr entgegen.



> Man könnte höchstens einfach den Namen ändern, da V1 ja inzwischen irrelevant ist. Dann kommen halt alle L3 ohne Hood und V1 in die Hood "No Hood"?

An irgendwie sowas hatte ich gedacht. Hauptsache nicht "Legacy". Noch besser wärs, wenn dann einfach keine Hood mehr auf der Statusseite angezeigt wird und in den Stats alles in einer "Others"-Gruppe landet.

**Comment by adschm** [comment]
Kann ich mit leben. Muss ich mir mal ansehen, wie man das am besten macht.

**Comment by ChristianDresel** [comment]
Da mich es immer wieder nervt das in der Layer 3 Firmware hoodname so doof zu setzen ist muss da jetzt mal eine Lösung her, z.t. wird es ja auch schon vergessen. Daher mein Vorschlag:

- v1 wird entfernt das Zeug haben wir jetzt lange genug mit rumgeschleppt und es werden ja auch schon keine Daten mehr ans Monitoring gesendet... Also weg damit, schluss Ende AUS!
- wird ein Hoodname mitgesendet, passiert das gleiche wie bisher. Abgleich mit keyxchange, gibt es die Hood dort ist es eine v2, gibt es die nicht ist es local
- Wird keine Hood mit gesendet ist es immer local, sprich wenn der User vergisst den Hoodname zu senden taucht der Router ganz normal als layer 3 (oder local...) Router auf

Man könnte jetzt noch überlegen ob man sich an das Namensschema anpasst, local in layer 3 unbenennen oder so?!

**Comment by adschm** [comment]
V1 kommt weg, okay. Dann brauchen wir nur noch einen Namen für Router ohne Hood. Sucht euch was aus.

In diesem Kontext: Das Monitoring ist zur Zeit ja eine wilde Mischung aus Englisch und Deutsch. Obwohl ich tendenziell ja ein Fan davon bin, Sachen gleich in Englisch aufzuziehen, generiert dies beim Monitoring eigentlich keinerlei Mehrwert:
* Unser gesamtes Netz liegt im deutschen Sprachraum
* Die vom Nutzer gesetzten Router-Daten sind ohnehin auf Deutsch
* Ich sehe keinen spezifischen Nutzen der Seite für Nicht-Deutschsprachler
* Wie bereits festgestellt sind einzelne Elemente der Seite ohnehin auf Deutsch

Aus den gleichen Argumenten macht es in meinen Augen auch wenig Sinn, den Mehraufwand für eine Mehrsprachenkonfiguration in Kauf zu nehmen. Ich überlege ernsthaft, die ganze Seite einfach an nem Nachmittag hart ins Deutsche zu übersetzen.

Dies hätte nebenbei den Vorteil, dass man die blöde "local hood" los wird, und von "dezentraler Hood" sprechen kann. Weiterhin bitte ich das bei der Antwort auf die einleitende Frage zu berücksichtigen (also, wie würde ein deutscher Default-Wert heißen?)

Ansonsten gerne hier einmal pro Woche Ping schreiben, damit ich die Sache nicht wieder vergesse.

**Comment by ChristianDresel** [comment]
> V1 kommt weg, okay. Dann brauchen wir nur noch einen Namen für Router ohne Hood. Sucht euch was aus.

Ich würde sie einfach in die gleiche Sparte wie Local Routers packen.

> 
> In diesem Kontext: Das Monitoring ist zur Zeit ja eine wilde Mischung aus Englisch und Deutsch. Obwohl ich tendenziell ja ein Fan davon bin, Sachen gleich in Englisch aufzuziehen, generiert dies beim Monitoring eigentlich keinerlei Mehrwert:
> 
>     * Unser gesamtes Netz liegt im deutschen Sprachraum
> 
>     * Die vom Nutzer gesetzten Router-Daten sind ohnehin auf Deutsch
> 
>     * Ich sehe keinen spezifischen Nutzen der Seite für Nicht-Deutschsprachler
> 
>     * Wie bereits festgestellt sind einzelne Elemente der Seite ohnehin auf Deutsch
> 
> 
> Aus den gleichen Argumenten macht es in meinen Augen auch wenig Sinn, den Mehraufwand für eine Mehrsprachenkonfiguration in Kauf zu nehmen. Ich überlege ernsthaft, die ganze Seite einfach an nem Nachmittag hart ins Deutsche zu übersetzen.

wäre ich dafür

> 
> Dies hätte nebenbei den Vorteil, dass man die blöde "local hood" los wird, und von "dezentraler Hood" sprechen kann. 

dafür

> Weiterhin bitte ich das bei der Antwort auf die einleitende Frage zu berücksichtigen (also, wie würde ein deutscher Default-Wert heißen?)

schwierige Frage, "dezentrale Router" gefällt mir jetzt auf Anhieb am besten. Aber ich kann mich auch mit vielen anderen anfreunden hauptsache das hat jetzt mal ein System und die ganzen Layer 3 Router in Nbg/Fü/Er landen im gleichen Layer ;)

> 
> Ansonsten gerne hier einmal pro Woche Ping schreiben, damit ich die Sache nicht wieder vergesse.

**Comment by adschm** [comment]
Ich würde trotzdem gerne den Fall, dass gar keine Hood übertragen wird, separat behandeln. Allein für die Statistik-Seite brauche ich ohnehin eine Kategorie. Wo man das dann auf die Karte malt, ist mir nicht so wahnsinnig wichtig.
Ich würde ggf. auch einfach Hood "Keine" verwenden.

Wenn man V1 komplett ausbaut, würden dann diese Router ohnehin bei den dezentralen in der Karte landen. "Dezentral" ist alles, was nicht in der V2-Hood-Liste steht. Die manuelle Sonderwurst für V1 würde man dann einfach entfernen.

**Comment by ChristianDresel** [comment]
Nein, keine Hood ist für mich das gleiche. Mittelfristig werde ich wohl alle meine Layer 3 Router keine Hood mehr mitschicken weil das eintragen einfach zu nervig ist und mMn unnötig ist. Für mich ist es daher wichtig, das ich auf der Karte einen Layer habe der nur(!) das Richtfunknetz anzeigt. Das ist eigentlich mittlerweile mein default Handgriff auf der Map das v1/v2 Zeug ausblenden und nur das RiFu Netz angucken und das muss auf einem Layer sein. Die ganzen Hotspots interessieren mich nicht. 

Also ja, ich würde v1 entfernen und alles was nicht mit dem keyxchange überein passt ist local/dezentral/OderWieImmerManEsNennt eingruppieren. Dann fallen da sowohl L3 Router mit Hood als auch ohne Hood rein und sie sind in einem Layer so wie ich es gerne hätte.

**Comment by fblaese** [comment]
> Also ja, ich würde v1 entfernen und alles was nicht mit dem keyxchange überein passt ist local/dezentral/OderWieImmerManEsNennt eingruppieren. Dann fallen da sowohl L3 Router mit Hood als auch ohne Hood rein und sie sind in einem Layer so wie ich es gerne hätte.

Ich denke genau das hatte Adrian vor.
Irgendwie müssen die Geräte intern ja behandelt werden. Wenn man das als "Keine" oder "None" einordnet, passt das denke ich. Zudem könnte man das dann ja theoretisch auch auf der Geräteseite ausblenden.

Wenn man mehrere Geräte (e.g. Access Points etc.) einer lokalen Hood im Monitoring platzieren will, dann kann die Angabe eines Hoodnamen u.U. schon Sinn machen.


> Aus den gleichen Argumenten macht es in meinen Augen auch wenig Sinn, den Mehraufwand für eine Mehrsprachenkonfiguration in Kauf zu nehmen. Ich überlege ernsthaft, die ganze Seite einfach an nem Nachmittag hart ins Deutsche zu übersetzen.

+1, solange Fachbegriffe erhalten bleiben, und nicht zwanghaft eingedeutscht werden (e.g. "Hood", "Firmware", etc.). Eigentlich bleibt dann aber kaum noch was übrig.

**Comment by ChristianDresel** [comment]
Hi

wie machen wir hier weiter? Es gibt weiterhin Layer 3 Router die keine Hood mehr gesetzt haben und ich will diese User auch darauf nicht anschreiben wenn wir das eh weg machen, führt nur zu Verwirrung ( https://monitoring.freifunk-franken.de/routers/7953 )

Und beim neu einrichten von Routern ist das bei mir auch jedes mal die Option die ich vergesse und dann fluche und wieder hier an diesen Beitrag denke ;).

Gruß
Christian
